Clamp TabController animation value when calc _DragAnimation.value

smadey opened the pull request:
## Description

see [demo](https://dartpad.dev/cf3e344cfdb5aa08a17c0d36aeaa0681)

## Tests

I added the following tests:

*TabController changes offset attribute*

## Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

No

smadey committed:
Clamp TabController animation value when calc _DragAnimation.value

smadey committed:
Add "TabController changes offset attribute" test

Piinks reviewed (commented):
I think this is meant to resolve https://github.com/flutter/flutter/issues/61156
@smadey is this the bug your demo is illustrating? I wasn't sure what this is supposed to be fixing.

smadey commented:
Not the same as #61156
See [gist](https://gist.github.com/smadey/cf3e344cfdb5aa08a17c0d36aeaa0681) or [repo](https://github.com/smadey/nest_tabs_demo)
Swipe left and right on the last outer tab and last inner tab to see the problem

![Simulator Screen Shot - iPhone X - 2020-07-20 at 20 13 35](https://user-images.githubusercontent.com/4351454/87936518-88295680-cac5-11ea-90b0-0f88324bfa36.png)

Piinks commented:
> Not the same as #61156
> See [gist](https://gist.github.com/smadey/cf3e344cfdb5aa08a17c0d36aeaa0681) or [repo](https://github.com/smadey/nest_tabs_demo)
> Swipe left and right on the last outer tab and last inner tab to see the problem
> 

Ah! Ok thanks for checking!
Can you add a short description to your PR explaining the bug? It's a bit hard without guidance to tell from the picture what the bug is that you are fixing or how to reproduce it from your sample code. Thanks! :)

smadey commented:
Swipe left and right on the last outer tab and last inner tab to see the problem
![RPReplay_Final1596186704](https://user-images.githubusercontent.com/4351454/88644698-df4aaf00-d0f5-11ea-9c3b-e99b49844567.gif)

Piinks commented on packages/flutter/test/material/tabs_test.dart:
```diff
@@ -2670,6 +2670,57 @@ void main() {
     final PageView pageView = tester.widget<PageView>(find.byType(PageView));
     expect(pageView.physics.toString().contains('ClampingScrollPhysics'), isFalse);
   });
+  
```
The analyzer is failing here due to a trailing whitespace.

Piinks reviewed (commented):
It looks like there is a merge conflict, can you update your branch?

smadey committed:
Merge branch 'master' into smadey-patch-1

godofredoc committed:
Remove hostonly_tests from cirrus as they are now running on LUCI. (#63257)

* Remove hostonly_tests from cirrus as they are now running on LUCI.

Bug:
  https://github.com/flutter/flutter/issues/63242

* Remove templates add framework tests for linux back.

a14n committed:
migrate painting to nullsafety (#63703)

* migrate painting to nullsafety

* address review comments

rami-a committed:
Mention time picker theme in showTimePicker documentation (#63776)

JuYeong0413 committed:
Remove enableR8=true from gradle.properties (#63339)

declanwoods committed:
Fix SafeArea and SliverSafeArea debug flag properties (#63455)

goderbauer committed:
Make textfields restorable (#63401)

goderbauer committed:
Use reportError (over onError) everywhere (#63784)

keyonghan committed:
Add enabled flag for try builders (#63783)

* add enabled flag

* remove flaky flag for try builders

* remove trailing space

stereotype441 committed:
Ignore/fix dead_code checks for weak-only null checks. (#63794)

Now that https://github.com/dart-lang/sdk/issues/41985 is fixed, the
analyzer detects dead code due to "unnecessary" null checks.  Since we
want to retain null checks in Flutter even when they appear
unnecessary (in order to preserve runtime behavior in weak mode), we
need to suppress these dead code hints.

Note that one assertion is removed by this PR (`heightFactor != null
|| (heightFactor == 1.0 && heightDelta == 0.0)`).  This is because
`heightFactor == 1.0 && heightDelta == 0.0` can only be true if
`heightFactor != null`, so this assertion is equivalent to simply
asserting that `heightFactor != null`, which is already asserted two
lines above.

zanderso committed:
[flutter_tool] Fix some create_test.dart tests on Windows (#63796)

nscobie committed:
Add viewport documentation breadcrumbs (#63192)

CareF committed:
bug fix for flutter_gallery__transition_perf(_with_semantics) (#63799)

* report failure for semantics test

* bug fix

LongCatIsLooong committed:
Prevent viewport.showOnScreen from scrolling the viewport if the specified Rect is already visible.  (#56413)

scheglov committed:
Provide the argument for a required parameter 'data'. (#63728)

digiter committed:
Add an infra issue template (#63730)

rrousselGit committed:
add benchmark for ChangeNotifier.notifyListeners (#62715)

dnfield committed:
drop xcpretty for tests (#63804)

CareF committed:
bug fix for pointer id in test gestures (#63816)

digiter committed:
Limit about to <200 chars in the infra issue template (#63839)

engine-flutter-autoroll committed:
Roll Engine from 7571e7c380fd to b300be3df32b (60 revisions) (#63892)

zichangg committed:
Remove TODOs added for HTTP abort() change in dart sdk (#63924)

Commit:
Revert "Remove TODOs added for HTTP abort() change in dart sdk (#63924)" (#63972)

This reverts commit b1c17313f6cbcd1f7ec083a943b28a076809eb0a.

Commit:
Revert "Roll Engine from 7571e7c380fd to b300be3df32b (60 revisions) (#63892)" (#63969)

This reverts commit b2c0970aa7b5334580c627316e2a07500c7d9c64.

scutlight committed:
Add debug flag `debugProfileLayoutsEnabled`. (#63652)

parkwonhui committed:
DecoratedBoxTransition sample code (#63775)

helin24 committed:
Return host and port from devtools launcher (#63795)

stuartmorgan-g committed:
Separate blkid in the Linux library checks (#63810)

LongCatIsLooong committed:
Prevent material switch from recreating its render object when it becomes disabled (#61398)

zanderso committed:
[flutter_tool] Handle Windows line endings in packages_test.dart (#63806)

engine-flutter-autoroll committed:
Roll Engine from 7571e7c380fd to 613be923b132 (76 revisions) (#63979)

* 94f5cf4b1 Roll Dart SDK from 1a1ab78dd6bc to 91e46dcbbbbd (4 revisions) (flutter/engine#20431)

* e0d515369 Make CkVertices a ManagedSkiaObject (flutter/engine#20421)

* 5161fd1a6 Roll Skia from 864562faf497 to b3f26441d191 (22 revisions) (flutter/engine#20447)

* 2ac5b3460 Revert "Refactor initial route code (#19684)" (flutter/engine#20450)

* 5c8a62608 Delete CSS paint code (flutter/engine#20426)

* 9f146b5f4 Add --unoptimied and --asan to Fuchsia builds (flutter/engine#20427)

* 4fb3ea81a Roll Dart SDK from 91e46dcbbbbd to abdfa80162c1 (3 revisions) (flutter/engine#20453)

* 4e2178742 Roll Skia from b3f26441d191 to c95b589112f3 (12 revisions) (flutter/engine#20456)

* d49c94b6e Roll Fuchsia Mac SDK from us_8V5gGK... to ik9kEVW-J... (flutter/engine#20457)

* 2eebe4a94 Roll Fuchsia Linux SDK from vomfYvIJk... to xcu8qvX33... (flutter/engine#20458)

* debb30edd Fix broken symbols on Fuchsia embedder (flutter/engine#20459)

* 2bc0bfdab Send locale information in the macOS embedding (flutter/engine#20461)

* 651cb384a Roll Skia from c95b589112f3 to 652124c372e5 (4 revisions) (flutter/engine#20462)

* 95b2ebfaf Roll Dart SDK from abdfa80162c1 to 279a4b920da8 (1 revision) (flutter/engine#20463)

* de6810b6a Roll Skia from 652124c372e5 to 025843fad003 (1 revision) (flutter/engine#20467)

* 52c1ec6a2 Roll Dart SDK from 279a4b920da8 to 00016747ab48 (2 revisions) (flutter/engine#20471)

* 7d45e22ea Roll Dart SDK from 00016747ab48 to 9ae3e22f1e53 (1 revision) (flutter/engine#20474)

* aa4601156 Roll Fuchsia Mac SDK from ik9kEVW-J... to c9oMk4qlD... (flutter/engine#20476)

* 4746a93ae Roll Skia from 025843fad003 to df9ed89229db (4 revisions) (flutter/engine#20478)

* dd2c5a1b8 Plumbing for setting domain network policy (flutter/engine#20218)

* f0955b071 Roll Fuchsia Linux SDK from xcu8qvX33... to AjjDqKkUV... (flutter/engine#20477)

* eef639ca2 Roll Skia from df9ed89229db to 8dd1e22f4130 (2 revisions) (flutter/engine#20481)

* f1855fb5a Clean-up CanvasKit canvas sizing. (flutter/engine#20387)

* d9f00a38a Roll Skia from 8dd1e22f4130 to af36652766db (6 revisions) (flutter/engine#20484)

* 4a91bc8de Roll Dart SDK from 9ae3e22f1e53 to 6d15e04622c2 (1 revision) (flutter/engine#20483)

* f3964186c Do not access members after calling ClearDartWrapper (flutter/engine#20465)

* dacd2d435 Roll Skia from af36652766db to cc01311cfc96 (4 revisions) (flutter/engine#20485)

* 9cd99d265 Add enabled flag for try builders (flutter/engine#20460)

* 0c504da03 Refactor initial route code take 2 (flutter/engine#20468)

* 6d44559b7 Remove the dummy rasterizer delegate now that flutter_runner is in tree, and cleanup ctor params (flutter/engine#20486)

* 16c328d12 Roll Skia from cc01311cfc96 to 9e77f20864d3 (6 revisions) (flutter/engine#20489)

* da3c9ff53 Fix unoptimized fuchsia builds (flutter/engine#20488)

* 9e3038371 Roll Skia from 9e77f20864d3 to 311dd9d8474c (4 revisions) (flutter/engine#20493)

* f644d0350 Add a service protocol for raster cache memory (flutter/engine#20466)

* 6685b3195 Roll Skia from 311dd9d8474c to 283c230fa18a (2 revisions) (flutter/engine#20499)

* d6052c001 Roll Fuchsia Linux SDK from AjjDqKkUV... to -dcNYnqRy... (flutter/engine#20502)

* 022058f7e Roll Fuchsia Mac SDK from c9oMk4qlD... to B75vjVVwe... (flutter/engine#20504)

* 9c99bf758 Add missing nullable annotation (flutter/engine#20505)

* d4f610065 Roll Skia from 283c230fa18a to b6ce6baf7a7a (1 revision) (flutter/engine#20506)

* 84aa36568 Fix the legacy EncodableValue codepaths (flutter/engine#20501)

* c3d03b7ad Roll Skia from b6ce6baf7a7a to c184cf36092f (3 revisions) (flutter/engine#20507)

* 5aaebfcff Roll Skia from c184cf36092f to 1cf2c8d6ec21 (1 revision) (flutter/engine#20509)

* 4899b1c69 Roll Skia from 1cf2c8d6ec21 to e219c7ed3754 (1 revision) (flutter/engine#20512)

* 0d83051b1 Roll Skia from e219c7ed3754 to c9c4e2e9efac (3 revisions) (flutter/engine#20513)

* 33015c6c7 hasStrings on Android (flutter/engine#20393)

* 7b1ee9fdf Roll Skia from c9c4e2e9efac to 166cd52cee64 (4 revisions) (flutter/engine#20515)

* b32a2a5b0 Make sure ios tests fail if they fail (flutter/engine#20518)

* 0cda2931e Fixed the accessibility ios unit tests by sending in mock view (flutter/engine#20522)

* 999322825 Fix web_ui warnings flagged by the next Dart analyzer update (flutter/engine#20524)

* 169b22cda Handle glfwGetPrimaryMonitor returning nullptr (flutter/engine#20523)

* 429675d31 Custom Gradle home and add .vpython file (flutter/engine#20425)

* 9333b7c1d Create PlatformView instance right after method channel call from Dart (flutter/engine#20500)

* f0ca3d29a Roll Fuchsia Linux SDK from -dcNYnqRy... to WXwJgZ7tK... (flutter/engine#20517)

* dbf7e8311 Roll Skia from 166cd52cee64 to 01d016186340 (9 revisions) (flutter/engine#20519)

* 5a2bf6aa4 Reland: "fuchsia: Remove dead code / break dependencies" (flutter/engine#20532)

* 1590fb23c Roll Fuchsia Mac SDK from B75vjVVwe... to tmHY285qa... (flutter/engine#20534)

* f60e83696 Started stubbing in the FlutterView property of the mock (flutter/engine#20542)

* ac6442a65 Roll Fuchsia Linux SDK from WXwJgZ7tK... to t6BpwE-Ie... (flutter/engine#20539)

* 7d4b549eb Roll Fuchsia Mac SDK from tmHY285qa... to 0ZIOLxuHh... (flutter/engine#20544)

* b300be3df Roll Dart SDK from 6d15e04622c2 to 302893b93170 (11 revisions) (flutter/engine#20545)

* 31a0e45eb Roll Skia from 01d016186340 to a008b0fa8bbb (14 revisions) (flutter/engine#20546)

* ac8175fee Only create raster_thread_merge when explicitly requested by the embedding platform (flutter/engine#20487)

* 8430fa1e6 Roll Fuchsia Linux SDK from t6BpwE-Ie... to TD-QSKxG8... (flutter/engine#20548)

* ddf253b4c Roll Fuchsia Mac SDK from 0ZIOLxuHh... to EEj2OHioI... (flutter/engine#20549)

* 7fa21a45a [windows] Expose the binary messenger from FlutterEngine (flutter/engine#20399)

* 3ebcde6a1 Revert "[windows] Expose the binary messenger from FlutterEngine (#20399)" (flutter/engine#20550)

* 3aff25692 Enable iOS plumbing for network security and add tests (flutter/engine#20492)

* f74e4277d Roll Dart SDK from 302893b93170 to 58a209093c38 (1 revision) (flutter/engine#20552)

* fb581b989 Roll Skia from a008b0fa8bbb to d882e93e416b (2 revisions) (flutter/engine#20553)

* 219fc839e Roll Skia from d882e93e416b to c4245f475be7 (2 revisions) (flutter/engine#20554)

* 2116ed2e1 Roll Fuchsia Mac SDK from EEj2OHioI... to ko0N4Pxl_... (flutter/engine#20555)

* 76d492229 Roll Fuchsia Linux SDK from TD-QSKxG8... to oNnqaiRXK... (flutter/engine#20556)

* 32b1b70e2 [windows] Expose the binary messenger from FlutterEngine (flutter/engine#20551)

* 9060913dc Send locale information in the Windows embedding (flutter/engine#20455)

* d7510dffa Roll Dart SDK from 58a209093c38 to e74e4de7a430 (1 revision) (flutter/engine#20559)

* 613be923b Roll Skia from c4245f475be7 to fdbd50a8dc87 (4 revisions) (flutter/engine#20561)

CareF committed:
add ios tests (#63797)

CareF committed:
doc typo spelling fix (#63817)

liyuqian committed:
Mark nonflaky tests as such (#63801)

Commit:
[flutter_tools] trim trailing newline from vm service logging events (#63976)

Commit:
Revert "Roll Engine from 7571e7c380fd to 613be923b132 (76 revisions) (#63979)" (#63991)

This reverts commit 1e36d44030104f3c52a013d91d684c8f62d4c7ad.

CareF committed:
ensureVisible => scrollUntilVisible (#63983)

engine-flutter-autoroll committed:
Roll Engine from 7571e7c380fd to fbfe03882667 (79 revisions) (#63993)

Hixie committed:
Build routes even less. (#62588)

CareF committed:
add vsync overhead to `FrameTimingSummarizer` (#63835)

engine-flutter-autoroll committed:
Roll Engine from fbfe03882667 to b9243713a097 (2 revisions) (#63998)

albertviilik committed:
Make deprecation notice link to website downloads (#63829)

pingbird committed:
Fix FittedBox BoxFit.scaleDown sizing (#63668)

Commit:
[flutter_tools] add cache busting query param (#63818)

engine-flutter-autoroll committed:
Roll Engine from b9243713a097 to 02de4eb7cefa (8 revisions) (#64013)

engine-flutter-autoroll committed:
Roll Engine from 02de4eb7cefa to 0b330467da24 (2 revisions) (#64024)

mehmetf committed:
Revert "fix wrap intrinsic height calculation (#63420)" (#64046)

This reverts commit 72619b86421bfdbf4ef7b98cb893b168b89c8677.

aam committed:
Move away from deprecated NoSuchMethodError to NoSuchMethodError.withInvocation (#63974)

Commit:
[flutter_tools] add more info to coverage script (#64065)

Also output covered | total lines, and a summary percentage at the end + Kick tree

mehmetf committed:
Revert "Prevent viewport.showOnScreen from scrolling the viewport if the specified Rect is already visible.  (#56413)" (#64091)

This reverts commit 64d76f2fb7ed5eb785c660084b18ed5a51fd926b.

mehmetf committed:
Revert "Ensure time picker input mode lays out correctly in RTL (#63599)" (#64094)

This reverts commit 1de673bcda547a36406b77efebdeb078b1a073d3.

gspencergoog committed:
Hook up soft keyboard "next" and "previous" buttons so that they move the focus by default (#63592)

Focus will be moved automatically if onEditingComplete is not specified, but must
by moved manually if onEditingComplete is specified.

Hixie committed:
Typo (#63692)

Commit:
[flutter_tools] correctly forward error only stdout in non-verbose modes (#63815)

Commit:
[flutter_tools] shrink fingerprinter API to currently used subset (#63840)

godofredoc committed:
Make cirrus tests presubmit only and add new luci tests. (#63995)

CareF committed:
remove duplicate e2e driver (#64015)

jmagman committed:
Use real flutter root for finding pub in command integration test (#64021)

goderbauer committed:
migrate semantics to nullsafety (#64055)

xster committed:
Incorrect path.join usage might be causing Windows test failures (#64056)

Commit:
[null-safety] enable null assertions for framework tests (#64071)

goderbauer committed:
Bump dartdoc to 0.30.4 (#64112)

digiter committed:
Fix a data issue in prod_builder.json (#64116)

jmagman committed:
Revert "[null-safety] enable null assertions for framework tests (#64071)" (#64118)

This reverts commit acdb909fcb7fa078d2c536db4c5182c4b1b37337.

goderbauer committed:
Fix assert in dartdoc_checker and enable asserts (#64111)

godofredoc committed:
Remove windows docs from list of try jobs. (#64125)

Bug:
  https://github.com/flutter/flutter/issues/64121

justinmc committed:
Document how to size IV's child correctly, after seeing confusion in Github issues (#64100)

LongCatIsLooong committed:
Restore the original name of the velocity tracker microbenchmark (#64060)

rami-a committed:
[Material] Relanding fix to ensure time picker input mode lays out correctly in RTL (#64097)

Commit:
[versions] update gallery, roll versions (#64067)

digiter committed:
[devicelab] Enable a LUCI try builder. (#64004)

mono0926 committed:
Fix title color of license detail page (#63249)

engine-flutter-autoroll committed:
Roll Engine from 0b330467da24 to dcbe3d3d3dff (25 revisions) (#64161)

liyuqian committed:
Mark tests as nonflaky (#64077)

These tests look quite stable for the last 10 runs.

Commit:
[flutter_tools] reland: enable --null-assertions for flutter framework tests (#64120)

Reland of acdb909 with fixes to semantics.dart

#61042

liyuqian committed:
Rename needsMeasureCpuGpu to measureCpuGpu (#64000)

rami-a committed:
Remove unused copyWith arguments in TimePickerThemeData (#64167)

mdebbar committed:
Make SelectableText work better on web (#63786)

darrenaustin committed:
Changes to initialDate for CalendarDatePicker should update the state. (#64115)

engine-flutter-autoroll committed:
Roll Engine from dcbe3d3d3dff to 81779371338a (4 revisions) (#64171)

Piinks reviewed (approved):
This LGTM! Thank you for the contribution! 
It looks like there is a trailing whitespace that is causing the analyzer to fail. Once that's fixed we can merge this! 🎉

justinmc committed:
ClipboardStatusNotifier should handle errors in Clipboard.getData (#64012)

zanderso committed:
WIP: [flutter_tool] Fix amberctl host address (#64020)

Commit:
[flutter_tools] shrink API surface of Android SDK (#63867)

Attempt to simplify the Android SDK interface ahead of refactoring it. The locateAndroidSdk static method is called at startup to locate the android SDK, returning null if it cannot be found. These helper methods attempted to first look up the AndroidSDK if it was already null - which could only cover the case where someone installed the Android SDK while flutter was running (possibly through an IDE)

digiter committed:
Reorder issue templates. (#64174)

Commit:
[flutter_tools] null assertions off by default for web (#64186)

Like Android/iOS, only enable --null-assertions if asked. Previously this was enabled by default for web, but in general this has proved to be too breaking to enable by default.

#61042

gspencergoog committed:
Standardize the nullability of arguments to raw key data constructors. (#63523)

This standardizes the handling of nullability for the LogicalKeyboardKey.keyLabel and RawKeyEventData.keyLabel accessors so that they are non-nullable, but can be empty.

Before this change, the keyLabel could be either null or an empty string to indicate that there wasn't a label, which makes it harder to test for, since both need to be checked for. Since an empty string is sufficient, there is no need for it to be nullable.

Also, in raw_keyboard.dart, the web and Windows implementations wouldn't accept null values for parameters in the Map coming from the message, but tests were supplying null for some of them. This makes web and Windows creation of events match the other platforms, and makes the migration of tests to non-nullability easier.

Commit:
[null-safety] roll to null-safe platform (#64160)

engine-flutter-autoroll committed:
Roll Engine from 81779371338a to 74d32ecdbd46 (2 revisions) (#64193)

jimmy-robert committed:
Fix tooltip position. (#60479)

chunhtai committed:
remove ununsed deprecated field (#64076)

Commit:
[flutter_tools] update dwds, roll to null-safe file and process (#64195)

Update to latest dwds, file-nullsafety, and process-nullsafety for #62886

tvolkert committed:
(insert|move|remove)ChildRenderObject Deprecation: Step 1 (#64189)

* (insert|move|remove)ChildRenderObject Deprecation: Step 1

This deprecates the following methods:

* RenderObjectElement.insertChildRenderObject
* RenderObjectElement.moveChildRenderObject
* RenderObjectElement.removeChildRenderObject

...and replaces them with the following methods:

* RenderObjectElement.insertRenderObjectChild
* RenderObjectElement.moveRenderObjectChild
* RenderObjectElement.removeRenderObjectChild

The reason for the deprecation is to provide the `oldSlot` argument to
the `moveRenderObjectChild` method (such an argument was missing from
the now-deprecated `moveChildRenderObject` method) and the `slot`
argument to the `removeRenderObjectChild` method (such an argument was
missing from the now-deprecated `removeChildRenderObject` method). While
no argument was added to `insertRenderObjectChild`, the name change (and
corresponding deprecation) was made to maintain naming parity with the
other two methods.

This initial step does not update or remove any of the `slotToChild`
patterns that exist in the framework. This work is being separated
into two commits in case something needs to be reverted to minimize
the scope of each commit.

See https://github.com/flutter/flutter/issues/63269 for more info

christopherfujino committed:
Don't fail docs-linux on release branches (#64203)

CareF committed:
run the test longer time (#64185)

dkwingsmt committed:
Change MouseTracker's interface for clarity. Simplify MouseRegion's implementation. (#64119)

* Redesigns the interface between MouseTracker and RendererBinding&RenderView.
* Simplifies the structure of RenderMouseRegion.
* Extracts the common utility code between mouse_tracker_test and mouse_tracker_cursor_test.

smadey committed:
Remove trailing whitespace before test "TabController changes offset attribute"

engine-flutter-autoroll committed:
Roll Engine from 74d32ecdbd46 to 89f506ed626b (10 revisions) (#64213)

thecalamiity committed:
Added tests (#62635)

xster committed:
Move v1 embedding migration warning from plugin consumers to all apps (#64181)

stereotype441 committed:
Ignore unnecessary casts that can go away soon. (#64200)

engine-flutter-autoroll committed:
81027ab0c Reland: Add android scenarios presubmit task. (#64218)

mehmetf committed:
Revert "(insert|move|remove)ChildRenderObject Deprecation: Step 1 (#64189)" (#64249)

This reverts commit 21cfed34d478d6a1c0223557a7935b69b15613bb.

tvolkert committed:
Re-apply "(insert|move|remove)ChildRenderObject Deprecation: Step 1 (#64189)" (#64254)

This reverts commit ce40de69b7b4f89c66d19c8dbd3bd86ae30f1bc6.

(this re-applies #64189 by reverting #64249)

engine-flutter-autoroll committed:
Roll Engine from 81027ab0cc0e to 4dc866283d13 (2 revisions) (#64229)

christopherfujino committed:
Restore docs linux post submit cirrus (#64214)

tvolkert committed:
Remove `childToSlot` pattern (#64217)

flar committed:
report list of detail files so devicelab can save them (#64009)

xster committed:
Move Android doctor SDK check to 29 and Cirrus images to 30 (#63517)

Hixie committed:
Remove more timeouts (#63433)

lidongze91 committed:
Add native stacktrace field for PlatformException. (#64199)

CareF committed:
update TimelineStream in `flutter_driver` (#64258)

CareF committed:
Revert "run the test longer time (#64185)" (#64255)

Piinks committed:
Fix floating snackbar Y (#64268)

liyuqian committed:
Mark e2e version of ios32 transition test nonflaky (#64269)

timekone committed:
Clip image of BoxDecoration to circle when shape is BoxShape.circle (#63352)

godofredoc committed:
Remove windows_docs. (#64277)

keyonghan committed:
Add run_if property for existing LUCI try builders (#64180)

* add run_if property for existing try builders

* update

* update readme

stuartmorgan-g committed:
Add WindowProc delegation to Windows template (#64166)

This wires up the new WindowProc delegation system that allows plugins
to handle top-level window messages (e.g., to control resize behavior).

Fixes #53168

stuartmorgan-g committed:
Update Windows template for wrapper file change (#64173)

* Minor adjustment for engine change, and rev template version

* whitespace fix

dnfield committed:
Make large jumpTo recommend deferred loading (#64271)

LongCatIsLooong committed:
Revert "Prevent material switch from recreating its render object when it becomes disabled (#61398)" (#64062)

tvolkert committed:
Remove remaining uses of `childToSlot` (#64273)

The remaining uses of this pattern were all due to wanting to have
the child's slot when `Element.forgetChild()` was called. However,
when that method is called, the child's `slot` value is still valid
in the context of the parent, so the uses can just use `child.slot`.

This is the final round of cleanup from the fallout of #63269

Commit:
[null-safety] remove mockito from network tests (#64284)

More work towards removal of mockito to facilitate strong null safety tests.

#62886

Hixie committed:
Trivial typo fixes to snippets documentation (#63436)

Hixie committed:
More restoration documentation (#63438)

Commit:
[flutter_tools] check for permission issues when copying file (#63540)

xu-baolin committed:
Improve docs for [PageStorage] (#63634)

Hixie committed:
Add null safety migration annotations. (#64227)

engine-flutter-autoroll committed:
Roll Engine from 4dc866283d13 to ae2dbefbedf8 (14 revisions) (#64303)

godofredoc committed:
Add web_smoke_tests to luci. (#64331)

Bug:
  https://github.com/flutter/flutter/issues/64070

Commit:
[null-safety] remove mockito from refresh control test (#64280)

engine-flutter-autoroll committed:
Roll Engine from ae2dbefbedf8 to d495da20d069 (7 revisions) (#64349)

Commit:
Increase threshold for usage of compute for utf8 decoding on large strings to 50 KB (#64350)

On a local benchmark, removing compute is faster than compute in both debug and release mode on a Pixel 4. On a MotoG4, a much larger text sample (800 Kb) takes 50 ms, so we cannot simple remove the limit. Increase to 50 Kb which should take at most a ms or two on an older device, and only microseconds on newerones.

LongCatIsLooong committed:
Reland "Prevent material switch from recreating its render object when it becomes disabled (#61398)", reverted in #64062 (#64354)

liyuqian committed:
Change typo profileDemos to restDemos (#64364)

digiter committed:
Fix LUCI builders config. (#64361)

Commit:
Revert "Increase threshold for usage of compute for utf8 decoding on large strings to 50 KB (#64350)" (#64372)

This reverts commit f07f4120e9a5232a7c7af3151f254d5536a79b44.

CareF committed:
migrate flutter_gallery_sksl_warmup__transition_perf to e2e (#64275)

Commit:
[devicelab] disable image list duration tests (#64486)

dreveman committed:
Pointer event resampler (#41118) (#60558)

Commit:
[null-safety] remove mockito from route test (#64363)

Removes more mockito to unblock migration of all tests to sound null safety. #62886

AlexV525 committed:
[EditableText] Fix TextField crashed with composing and maxLength set (#63754)

0xapurv committed:
fix: typo fixed from ENTER to SPACE (#64235)

LongCatIsLooong committed:
Add rect logic to getOffsetToReveal when onlySlivers is true (#64295)

engine-flutter-autoroll committed:
Roll Engine from d495da20d069 to aa08192a6bef (14 revisions) (#64393)

tvolkert committed:
Pass --verbose-logging to flutter_tester (#64440)

tvolkert committed:
Add GestureDetector.onDoubleTapDown() (#64431)

* Add GestureDetector.onDoubleTapDown()

* Review comments

liyuqian committed:
Mark new_gallery__crane_perf as flaky (#64341)

HansMuller committed:
Removed deprecated ContainedButton and ContainedButtonTheme classes (#64483)

HansMuller committed:
Update origin material button API doc with obsolete API warnings (#64472)

engine-flutter-autoroll committed:
Roll Engine from aa08192a6bef to 4bb6251015a2 (6 revisions) (#64496)

Commit:
Revert "Pass --verbose-logging to flutter_tester (#64440)" (#64517)

This reverts commit 8b52e6a8b2994e848ff43841c8bad154237dc0e8.

Commit:
[flutter_tools] remove globals from utils (#63831)

Commit:
[flutter_tools] migrate flutter_goldens, flutter_goldens client to null-safety (#64201)

xu-baolin committed:
Fix bug when tapping ListTitle with CheckboxListTile  tristate enable (#63925)

* fix bug when tap ListTitle when tristate enable #63846

* fix bug when tap ListTitle when tristate enable #63846

* code style

* improve the unit test case

* bow to convention

engine-flutter-autoroll committed:
Roll Engine from 4bb6251015a2 to 4caee915230b (4 revisions) (#64518)

Commit:
Reland: Increase threshold for usage of compute for utf8 decoding on large strings to 50 KB (#64498)

engine-flutter-autoroll committed:
Roll Engine from 4caee915230b to b792de7baea2 (3 revisions) (#64523)

jmagman committed:
Change iOS plugin simulator archs from allow list to deny list (#64504)

LongCatIsLooong committed:
Warns when single line text fields overflow (#63639)

engine-flutter-autoroll committed:
Roll Engine from b792de7baea2 to 8107b36c7b13 (2 revisions) (#64525)

mdebbar committed:
[web] Fix and enable editable_text_test.dart (#64516)

stuartmorgan-g committed:
[windows] Update plugin template (#64521)

The pointer-argument-based version of Success() is deprecated; this
updates the template to use the new reference-based version (which
allows for inline construction of the response value).

Part of https://github.com/flutter/flutter/issues/63975

Commit:
[flutter_tools] support code size tooling on iOS, linux, windows, macOS, and Android on Windows (#63610)

Adds support for size analysis on iOS, macOS, linux, and Windows - using an uncompressed directory based approach. The output format is not currently specified.

Adds support for size analysis on android on windows, switching to package:archive

Updates the console format to display as a tree, allowing longer paths. Increases the number of dart libraries shown (to avoid only ever printing the flutter/dart:ui libraries, which dominate the size)

chunhtai committed:
Reland "fix wrap intrinsic height calculation (#63420)" (#64276)

Commit:
[flutter_tools] refactor artifact downloading to retry zip exceptions. (#64512)

CareF committed:
update manifest (#64079)

jmagman committed:
Bump Cirrus image Xcode version (#63252)

Commit:
[flutter_tools] handle FileSystemException thrown when attempting to decode stamp file (#64559)

Fixes #64246

If decoding the string contents of a stamp file fails, treat it as if the stamp file is missing. This is always safe to do, since at worst it downloads more artifacts. The files could end up corrupted if they were edited by another program, for example

Commit:
[flutter_tools] avoid creating Android Devices if AndroidSDK cannot be found (#64524)

Avoid creating AndroidDevice discovery if the SDK cannot be located. Previously the tool would use which/where adb, however this required us to handle the AndroidSdk class being potentially null - which required an additional layer of indirection around all access. Sometimes these were forgotten leading to NPEs.

In general, not much can be done with an Android Device if the actual SDK is not installed.

LongCatIsLooong committed:
Revert "Warns when single line text fields overflow (#63639)" (#64573)

This reverts commit 0ab5ecc86b1f13854ecc0cb4a89fbdc9938c5212.

aam committed:
Re-enable image_list test with updated certificate(good for 3650 days). (#64519)

Added instructions on how to regenerate certificate too.

gspencergoog committed:
Protect against null context in release mode (#64474)

engine-flutter-autoroll committed:
Roll Engine from 8107b36c7b13 to d241105d7a84 (5 revisions) (#64580)

Commit:
[null-safety] remove sound-null-safety by default, remove experiment flag (#64594)

To allow tests to begin migrating to null safety, sound null safety needs to be disabled. Also removes the experiment flag that is unnecessary since the framework is on the SDK allowlist.

engine-flutter-autoroll committed:
Roll Engine from d241105d7a84 to 4a88d5e1099d (3 revisions) (#64593)

engine-flutter-autoroll committed:
0f80fff35 Roll Skia from 578215be44d0 to 5660e8b4414b (7 revisions) (flutter/engine#20749) (#64597)

jmagman committed:
Revert "[flutter_tools] avoid creating Android Devices if AndroidSDK cannot be found (#64524)" (#64602)

This reverts commit 47596c62038599045977fae87c98b25eab456379.

engine-flutter-autoroll committed:
Roll Engine from 0f80fff356b0 to b1d9f863db32 (3 revisions) (#64600)

dnfield committed:
Fix tests that use bogus objects, update TestImageInfo interface (#64599)

Hixie committed:
Migrate physics tests to NNBD (#64584)

engine-flutter-autoroll committed:
Roll Engine from b1d9f863db32 to c7467a4359b2 (9 revisions) (#64619)

kenzieschmoll committed:
Fix typo in flutter_tools --analyze-size output. (#64592)

engine-flutter-autoroll committed:
8ebeb0310 Roll Fuchsia Mac SDK from odfB5AW4C... to 2uJQ4vXzv... (flutter/engine#20776) (#64625)

dnfield committed:
Use Pixel4 for better wait times (#64609)

* Use Pixel4 for better wait times

According to the firebase team, these have lower wait times for tests.

Commit:
Revert "Reland: Increase threshold for usage of compute for utf8 decoding on large strings to 50 KB (#64498)" (#64652)

This reverts commit 54e2a0e6944ac04a9fb4b4b3fe96281daeee93b4.

rrousselGit committed:
Use a LinkedList to improve the performance of ChangeNotifier (#62330)

* Use a LinkedList in ChangeNotifier implementation to take O(N^2) notification time to O(N)

engine-flutter-autoroll committed:
Roll Engine from 8ebeb03103af to bb9dfcfc0dba (3 revisions) (#64658)

Piinks commented:
Hey @smadey would you be willing to update your branch once more? I think that will resolve the failing check here.

abrush21 committed:
Disables tests failing due to new semantics flag. (#63595)

engine-flutter-autoroll committed:
Roll Engine from bb9dfcfc0dba to 388193a67b6b (5 revisions) (#64667)

* b08c6b95e Fixing synthesizing keys for multiple keys pressed down on flutter web (flutter/engine#19632)

* 8308b6ad8 Avoid passing nil to IOS accessibility announcement  (flutter/engine#20700)

* 950b6a030 Roll Skia from ead4ca052b99 to 5da7327358e3 (1 revision) (flutter/engine#20782)

* 87fd0e497 Guard recording_canvas against restore calls without ending recording (flutter/engine#20786)

* 388193a67 Add tests for lerpDouble (flutter/engine#20778)

haeseoklee committed:
Fixed the description of the focusColor property included in input_decorator (#63981)

younghwan committed:
Update Visibility docs on maintainSize (#64148)

mdebbar committed:
[web] Enable word boundary tests (#64570)

christopherfujino committed:
remove docker_builder step (#64654)

CaseyHillers committed:
Sync with engine run_tests script (#64670)

smadey commented:
close by #64715
